tapdb/sqlc/migrations/000013_assets_unique_constraint.up.sql

```diff
@@ -0,0 +1,31 @@
+-- We will apply a new unique constraint on the assets table. This constraint
+-- will be on the columns (anchor_utxo_id, genesis_id, script_key_id).
+-- Rows in the existing table may violate this constraint, so we need to delete
+-- all but one row in each violating set of rows before applying the constraint.
+WITH duplicate_rows AS (
+    SELECT
+        asset_id,
+        anchor_utxo_id,
+        genesis_id,
+        script_key_id,
+
+        -- This is the row number of the row within the set of rows that violate
+        -- the constraint. We will delete all rows with a row number greater
+        -- than 1.
+        ROW_NUMBER() OVER (PARTITION BY anchor_utxo_id, genesis_id, script_key_id ORDER BY asset_id) AS row_num
+    FROM assets
+)
+DELETE FROM assets
+WHERE (anchor_utxo_id, genesis_id, script_key_id) IN (
+    SELECT anchor_utxo_id, genesis_id, script_key_id
+    FROM duplicate_rows
+    -- Delete all rows with a row number greater than 1. This should leave
+    -- exactly one row (row number 0) for each set of rows that violate the
+    -- constraint.
+    WHERE row_num > 1
+);
+
+-- Create a unique index on the new table
+CREATE UNIQUE INDEX assets_uniqueness_index_anchor_utxo_id_genesis_id_script_key_id
```

Comment:
Do we have a way to remove the index created by the previous `UNIQUE(asset_id, genesis_id, script_key_id)` part? I guess that's probably dependent on the actual DB implementation.
This is likely a downside of using the "inline" unique key creation, that it doesn't have a name and therefore can't easily be updated.

Reply:
Hmm, I think I google that before at some point... And if I remember correctly, we basically need to re-create the table to get rid of the original unique constraint. But I think we'd probably want to do that, as otherwise this is an expensive but useless index we're carrying along.
IIRC this is how you replace a table "correctly":
1. `CREATE TABLE assets_tmp ... (full definition, but without the inline UNIQUE constraint)`
2. `INSERT INTO assets_tmp AS SELECT * from assets`
3. Alter any foreign keys that point to the old `assets` table
4. `DROP TABLE assets`
5. `ALTER TABLE assets_tmp RENAME TO assets`